src/bio/bio_monitor.c

```diff
@@ -281,27 +281,68 @@ get_spdk_identify_ctrlr_completion(struct spdk_bdev_io *bdev_io, bool success,
 }
 
 static void
-populate_health_stats(struct spdk_nvme_health_information_page *page,
-		      struct nvme_stats *dev_state)
+populate_health_stats(struct bio_dev_health *bdh)
 {
-	union spdk_nvme_critical_warning_state	cw = page->critical_warning;
-
-	dev_state->warn_temp_time = page->warning_temp_time;
-	dev_state->crit_temp_time = page->critical_temp_time;
-	dev_state->ctrl_busy_time = page->controller_busy_time[0];
-	dev_state->power_cycles = page->power_cycles[0];
-	dev_state->power_on_hours = page->power_on_hours[0];
-	dev_state->unsafe_shutdowns = page->unsafe_shutdowns[0];
-	dev_state->media_errs = page->media_errors[0];
+	struct spdk_nvme_health_information_page	*page;
+	struct nvme_stats				*dev_state;
+	union spdk_nvme_critical_warning_state		cw;
+
+	page		= bdh->bdh_health_buf;
+	cw		= page->critical_warning;
+	dev_state	= &bdh->bdh_health_state;
+
+	/** commands */
+	d_tm_set_counter(bdh->bdh_du_written, page->data_units_written[0]);
+	d_tm_set_counter(bdh->bdh_du_read, page->data_units_read[0]);
+	d_tm_set_counter(bdh->bdh_write_cmds, page->host_write_commands[0]);
+	d_tm_set_counter(bdh->bdh_read_cmds, page->host_read_commands[0]);
+	dev_state->ctrl_busy_time	= page->controller_busy_time[0];
+	d_tm_set_counter(bdh->bdh_ctrl_busy_time,
+			 page->controller_busy_time[0]);
+	dev_state->media_errs		= page->media_errors[0];
+	d_tm_set_counter(bdh->bdh_media_errs, page->media_errors[0]);
+
+	dev_state->power_cycles		= page->power_cycles[0];
+	d_tm_set_counter(bdh->bdh_power_cycles, page->power_cycles[0]);
+	dev_state->power_on_hours	= page->power_on_hours[0];
+	d_tm_set_counter(bdh->bdh_power_on_hours, page->power_on_hours[0]);
+	dev_state->unsafe_shutdowns	= page->unsafe_shutdowns[0];
+	d_tm_set_counter(bdh->bdh_unsafe_shutdowns,
+			 page->unsafe_shutdowns[0]);
+
+	/** temperature */
+	dev_state->warn_temp_time	= page->warning_temp_time;
+	d_tm_set_counter(bdh->bdh_temp_warn_time, page->warning_temp_time);
+	dev_state->crit_temp_time	= page->critical_temp_time;
+	d_tm_set_counter(bdh->bdh_temp_crit_time, page->critical_temp_time);
+	dev_state->temperature		= page->temperature;
+	d_tm_set_counter(bdh->bdh_temp_crit_time, page->temperature);
```

Comment:
Looks like you're setting the wrong counter here. Should be `bdh_temp`

Reply:
Good catch!